dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -2823,6 +2866,189 @@ async fn cmd_db_sleds(
     Ok(())
 }
 
+// INSTANCES
+
+/// Run `omdb db instance info`: show details about a customer VM.
+async fn cmd_db_instance_info(
+    _: &OpContext,
+    datastore: &DataStore,
+    _: &DbFetchOptions,
+    args: &InstanceInfoArgs,
+) -> Result<(), anyhow::Error> {
+    use nexus_db_model::InstanceRuntimeState;
+    use nexus_db_queries::db::datastore::instance::InstanceGestalt;
+    let InstanceInfoArgs { id } = args;
+
+    // Destructure this exhaustively so that the OMDB command breaks and must be
+    // updated if new fields are added.
+    let InstanceGestalt { instance, active_vmm, target_vmm, migration } =
+        // use the super-special unauthorized version of this method just for OMDB
+        datastore
+            .instance_fetch_all_on_connection(
+                &*datastore.pool_connection_for_tests().await?,
+                &id,
+            )
+            .await
+            .with_context(|| {
+                format!("failed to fetch details for instance {id}")
+            })?;
+
+    // Manually format the instance struct because using its `fmt::Debug` impl
+    // will print out the user-data array one byte per line, which is horrible.
+    //
+    // /!\ WARNING /!\
+    // This does mean that anyone who adds new fields to the
+    // `nexus_db_model::Instance` type will want to make sure to update this
+    // code as well. Unfortunately, we can't just destructure the struct here to
+    // make sure this code breaks, since the `identity` field isn't public.
+    // So...just don't forget to do that, I  guess.
+    const ID: &'static str = "ID";
+    const NAME: &'static str = "name";
+    const DESCRIPTION: &'static str = "description";
+    const CREATED: &'static str = "created at";
+    const VCPUS: &'static str = "vCPUs";
+    const MEMORY: &'static str = "memory";
+    const HOSTNAME: &'static str = "hostname";
+    const AUTO_RESTART: &'static str = "auto-restart policy";
+    const STATE: &'static str = "state";
+    const LAST_MODIFIED: &'static str = "last modified at";
+    const LAST_UPDATED: &'static str = "last updated at";
+    const ACTIVE_VMM: &'static str = "active VMM ID";
+    const TARGET_VMM: &'static str = "target VMM ID";
+    const MIGRATION_ID: &'static str = "migration ID";
+    const UPDATER_LOCK: &'static str = "updater lock";
+    const ACTIVE_VMM_RECORD: &'static str = "active VMM record";
+    const MIGRATION_RECORD: &'static str = "migration record";
+    const TARGET_VMM_RECORD: &'static str = "target VMM record";
+    const WIDTH: usize = 1 + // breathing room
+        crate::helpers::const_max_len(&[
+        ID,
+        NAME,
+        DESCRIPTION,
+        CREATED,
+        VCPUS,
+        MEMORY,
+        HOSTNAME,
+        AUTO_RESTART,
+        STATE,
+        LAST_UPDATED,
+        LAST_MODIFIED,
+        ACTIVE_VMM,
+        TARGET_VMM,
+        MIGRATION_ID,
+        UPDATER_LOCK,
+    ]);
+    println!("\nINSTANCE\n");
+    println!("{ID:>WIDTH$}: {}", instance.id());
```

Comment:
Any value in including the project ID here as well? (The only other missing field seems to be the cloud-init user data, but I think omitting that makes sense; there's not a lot that can be done with it from this end of the system, and if we're debugging a guest issue with the config we can always just ask the issue reporter to provide that config.)

Reply:
Whoops, I meant to include the project ID, good catch. And, yeah, I had really wanted to just use the `fmt::Debug` representation of the `Instance` record as well, but realized that the cloud-init user data got formatted as a `Vec` of decimal numbers with each byte on its own line, which made the output more or less completely unreadable. So, all the custom formatting here is basically in service of excluding it.

Reply:
Fixed in https://github.com/oxidecomputer/omicron/pull/6610/commits/e3013e784ef9c7039a0b3e9b141531d4794ce629. :)